src/integer/poly_over_z/from.rs

```diff
@@ -68,8 +70,8 @@ impl FromStr for PolyOverZ {
         // We only have to check it once, because for every other position it checks
         // whether there is only one space.
         if !s_trimmed.contains("  ") {
-            return Err(MathError::InvalidStringToPolyMissingWhitespace(
-                s.to_owned(),
+            return Err(MathError::StringConversionError(
+                StringConversionError::InvalidStringToPolyMissingWhitespace(s.to_owned()),
             ));
```

Comment:
This looks prettier, please use it for the rest :)
```suggestion
            return Err(StringConversionError::InvalidStringToPolyMissingWhitespace(s.to_owned()))?;
```